spec/index.md

```diff
@@ -1289,6 +1293,18 @@ While an implementation will likely try to keep the interval between the value o
 
 :::
 
+#### On Low Wasm Memory {#on-low-wasm-memory}
+
+A canister can export a function with the name `canister_on_low_wasm_memory`, which is scheduled whenever the canister's wasm memory size in bytes grows from below a threshold `t` to >= `t`.
+The threshold `t` can be defined in the [canister's settings](#ic-canister_status) and by default it is set to 2<sup>64</sup> − 1.
+
+:::note
+
+While the above function is scheduled immediately once the condition above is triggered, it may not necessarily be executed immediately if it does not have enough cycles.
+If the canister gets frozen immediately after the function is scheduled for execution, the function will run once the canister's unfrozen _if_ the canister's wasm memory remains above the threshold `t`.
```

Comment:
So you don't plan to treat it analogously to call_on_cleanup after all?

Reply:
We discussed it and preferred to move forward with the scheduling approach. The reason for that is that `call_on_cleanup`, unlike this hook, cannot make outbound calls. Making outbound calls in the hook while also treating it like `call_on_cleanup` causes unnecessary complications in the implementation.

Reply:
> The reason for that is that call_on_cleanup, unlike this hook, cannot make outbound calls.

As long as it can schedule a one-off timer, I don't see a major issue with this limitation.